src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -122,4 +127,32 @@ protected static <S> S executeSupplierIfConcurrentModificationException(
     }
     return onOtherExceptionSupplier.get();
   }
+
+  /**
+   * This method calls the optional error callback specified in the {@code syncOptions} and updates
+   * the {@code statistics} instance by incrementing the total number of failed resources S to sync.
+   *
+   * @param errorMessage The error message describing the reason(s) of failure.
+   * @param exception The exception that called caused the failure, if any.
+   * @param oldResource the commercetools resource which could be updated.
+   * @param newResourceDraft the commercetools resource draft where we get the new data.
+   * @param updateActions the update actions to update the resource with.
+   * @param failedTimes The number of times that the failed cart discount statistic counter is
+   *     incremented.
+   */
+  @SuppressWarnings("unchecked")
+  protected void handleError(
+      @Nonnull final String errorMessage,
+      @Nullable final Throwable exception,
+      final S oldResource,
+      final T newResourceDraft,
+      List<UpdateAction<S>> updateActions,
+      final int failedTimes) {
+    SyncException syncException =
```

Comment:
```suggestion
    final SyncException syncException =
```

Reply:
Should be effectively final 😅 https://github.com/commercetools/commercetools-sync-java/pull/850/commits/879ca67ed21444c1152bed3dac4dc37ab7326662